packages/netlify-cms-widget-markdown/src/serializers/remarkShortcodes.js

```diff
@@ -10,17 +10,56 @@ export function remarkParseShortcodes({ plugins }) {
 
 function createShortcodeTokenizer({ plugins }) {
   return function tokenizeShortcode(eat, value, silent) {
-    let match;
-    const potentialMatchValue = value.split('\n\n')[0].trimEnd();
-    const plugin = plugins.find(plugin => {
-      match = value.match(plugin.pattern);
+    // Plugin patterns may rely on `^` and `$` tokens, even if they don't
+    // use the multiline flag. To support this, we fall back to searching
+    // through each line individually, trimming trailing whitespace and
+    // newlines, if we don't initially match on a pattern. We keep track of
+    // the starting position of each line so that we can sort correctly
+    // across the full multilen matches.
+    const trimmedLines = value
```

Comment:
I think we can simplify this function, I'll add another commit for it.
I also believe `start: 0` should be `start: -2` otherwise all the the `start` values will be shifted by `2`.

Reply:
See https://github.com/netlify/netlify-cms/pull/5597/commits/c394df846abd9cb799a272b3ddacb2b204b1867c.

I tried using a regex to get the indices, but that came out ugly